### PR TITLE
STORM-2220 Added passing cassandra config to individual Cassandra bolts instead of at topology level.

### PR DESCRIPTION
**Pull request by satishd**
More details at [STORM-2220](https://issues.apache.org/jira/browse/STORM-2220)

### CODE REVIEW

File: external/storm-cassandra/src/main/java/org/apache/storm/cassandra/bolt/BaseCassandraBolt.java [hunk 1]
```diff
@@ -105,17 +109,43 @@ public BaseCassandraBolt withResultHandler(ExecutionResultHandler resultHandler)
         return this;
     }
 
+    /**
+     * Configures this bolt with the given {@code fields} as outputfields with stream id as {@link Utils#DEFAULT_STREAM_ID}
+     *
+     * @param fields outputfields
+     */
     public BaseCassandraBolt withOutputFields(Fields fields) {
+        Preconditions.checkNotNull("fields should not be null.");
```
Comment (vesense) [inline external/storm-cassandra/src/main/java/org/apache/storm/cassandra/bolt/BaseCassandraBolt.java]: Missing args, to -> `Preconditions.checkNotNull(fields, "fields should not be null.");`

File: external/storm-cassandra/src/main/java/org/apache/storm/cassandra/bolt/BaseCassandraBolt.java [hunk 2]
```diff
@@ -92,6 +94,8 @@ public void prepare(Map stormConfig, TopologyContext topologyContext, OutputColl
         this.outputCollector = outputCollector;
         this.stormConfig = stormConfig;
         this.cassandraConfConfig = new CassandraConf(stormConfig);
+
+        if(configKey != null)
         this.client = clientProvider.getClient(this.stormConfig);
```
Comment (vesense) [inline external/storm-cassandra/src/main/java/org/apache/storm/cassandra/bolt/BaseCassandraBolt.java]: Maybe we need something like this? ->
```
if(configKey != null) {
    Map<String, Object> conf = (Map<String, Object>)map.get(this.configKey);
    this.client = clientProvider.getClient(conf);
} else {
    this.client = clientProvider.getClient(this.stormConfig);
}
```

Comment (satishd) [inline external/storm-cassandra/src/main/java/org/apache/storm/cassandra/bolt/BaseCassandraBolt.java]: Right, somehow the changes I see in my IDE and saved file in disk are different. I will update with the right sources. Thanks for raising this.

**Comment by harshach** [comment]
+1

File: external/storm-cassandra/src/main/java/org/apache/storm/cassandra/bolt/BaseCassandraBolt.java [hunk 3]
```diff
@@ -91,8 +93,12 @@ public BaseCassandraBolt(CQLStatementTupleMapper tupleMapper) {
     public void prepare(Map stormConfig, TopologyContext topologyContext, OutputCollector outputCollector) {
         this.outputCollector = outputCollector;
         this.stormConfig = stormConfig;
-        this.cassandraConfConfig = new CassandraConf(stormConfig);
-        this.client = clientProvider.getClient(this.stormConfig);
+
+        Map<String, Object> cassandraClientConfig = cassandraConfig != null ? stormConfig : stormConfig;
```
Comment (HeartSaVioR) [inline external/storm-cassandra/src/main/java/org/apache/storm/cassandra/bolt/BaseCassandraBolt.java]: This always pick stormConfig which seems not intended.

Comment (satishd) [inline external/storm-cassandra/src/main/java/org/apache/storm/cassandra/bolt/BaseCassandraBolt.java]: @HeartSaVioR Thanks for catching that.

**Comment by HeartSaVioR** [comment]
Looks good. +1

**Comment by ptgoetz** [comment]
+1

**Comment by satishd** [comment]
This PR is merged into master and 1.x-branch.

**Pull request closed without merging**